kong/templates/nginx_kong.lua

```diff
@@ -63,7 +64,10 @@ upstream kong_upstream {
 
 server {
     server_name kong;
-    listen ${{PROXY_LISTEN}}${{PROXY_PROTOCOL}};
+> for i = 1, #proxy_listeners do
+>   if proxy_listeners[i].ssl then _ssl_in_use = true end
```

Comment:
This logic should belong to our Lua logic prefix handling instead (passed when we compute the template), so that we keep a better separation of concern between our logic and this template, we should have no logic of its own.